Algorithm.Framework/Selection/UncorrelatedToBenchmarkUniverseSelectionModel.py

```diff
@@ -0,0 +1,112 @@
+# QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+# Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from clr import AddReference
+AddReference("QuantConnect.Algorithm")
+AddReference("QuantConnect.Algorithm.Framework")
+AddReference("QuantConnect.Common")
+
+from QuantConnect import *
+from QuantConnect.Algorithm import *
+from QuantConnect.Algorithm.Framework import *
+from Selection.FundamentalUniverseSelectionModel import FundamentalUniverseSelectionModel
+
+class UncorrelatedToBenchmarkUniverseSelectionModel(FundamentalUniverseSelectionModel):
+    '''This universe selection model picks stocks that currently have their correlation to a benchmark deviated from the mean.'''
+
+    def __init__(self,
+                 benchmark = Symbol.Create("SPY", SecurityType.Equity, Market.USA),
+                 numberOfSymbolsCoarse = 400,
+                 numberOfSymbols = 10,
+                 windowLength = 5,
+                 historyLength = 25):
+        '''Initializes a new default instance of the OnTheMoveUniverseSelectionModel
+        Args:
+            benchmark: Symbol of the benchmark
+            numberOfSymbolsCoarse: Number of coarse symbols
+            numberOfSymbols: Number of symbols selected by the universe model
+            windowLength: Rolling window length period for correlation calculation
+            historyLength: History length period'''
+        super().__init__(False)
+
+        self.benchmark = benchmark 
+        self.numberOfSymbolsCoarse = numberOfSymbolsCoarse
+        self.numberOfSymbols = numberOfSymbols
+        self.windowLength = windowLength
+        self.historyLength = historyLength
+        
+        # Symbols in universe
+        self.symbols = []
+        self.coarseSymbols = []
+        self.cor = None
+
+    def SelectCoarse(self, algorithm, coarse):
+
+        if not self.coarseSymbols:
```

Comment:
With this condition, the Coarse Universe data is only taken into consideration once. We need to change this behavior because of the dependency on the start date that it creates. I would suggest a monthly update used in many other models.